checker/src/main/java/org/checkerframework/checker/determinism/DeterminismAnnotatedTypeFactory.java

```diff
@@ -415,31 +421,68 @@ private void defaultArrayComponentTypeAsPolyDet(AnnotatedTypeMirror type) {
             // For example: if 'annoannoArrType' is "@Det int @Det[]",
             // "arrParamType.getExplicitAnnotations().size()" returns 0,
             // "arrParamType.getAnnotations().size()" returns 1.
-            if (annoArrType.getAnnotations().isEmpty()) {
-                recursiveDefaultArrayComponentTypeAsPolyDet(annoArrType);
+            if (annoArrType.getAnnotations().isEmpty() || annoArrType.hasAnnotation(NONDET)) {
```

Comment:
Why is the check `annoArrType.hasAnnotation(NONDET)` necessary? Is this for arrays explicitly annotated as @NonDet? A clarifying comment would be helpful.

Reply:
It's because we also check if `@NonDet` is explicitly written. We use `hasAnnotation` rather than `hasExplicitAnnotation` for the same reason we use `getAnnotations().isEmpty()` rather than `getExplicitAnnotation().isEmpty()`. I added this to the above comment.